SSLeay.xs

```diff
@@ -6283,8 +6283,28 @@ RSA_generate_key(bits,e,perl_cb=&PL_sv_undef,perl_data=&PL_sv_undef)
 void
 RSA_get_key_parameters(rsa)
 	    RSA * rsa
+PREINIT:
+#if defined(LIBRESSL_VERSION_NUMBER) && (LIBRESSL_VERSION_NUMBER >= 0x3050000fL)
+    BIGNUM *n, *e, *d;
+    BIGNUM *p, *q;
+    BIGNUM *dmp1, *dmq1, *iqmp;
```

Comment:
What if these are decleared as `const BIGNUM` and `bn2sv` is updated to `SV* bn2sv(const BIGNUM* p_bn)`? I think this would allow removing the casts when calling the `RSA_get0_*` functions.

Reply:
Good idea.  Works fine for LibreSSL.  Question is whether BN_dup() accepts a const BIGNUM for old versions of OpenSSL.  I will update the pull request, then CI will tell us.

Reply:
Looks good. I think going from non-const to const works, but not the other way around, and CI logs seem to confirm it.